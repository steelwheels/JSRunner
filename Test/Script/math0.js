/*
 * math0.js
 */

require('Math/Math') ;

console.log("Test: randomInt ... ") ;

let result = true ;
for(let i=0 ; i<11 ; i++){
	const v = Math.randomInt(0, 10) ;
	if(!(0<=v && v<=10)){
		result = false ;
	}
}

if(result){
	console.log("OK\n") ;
	Process.exit(0) ;
} else {
	console.log("NG\n") ;
	Process.exit(1) ;
}


