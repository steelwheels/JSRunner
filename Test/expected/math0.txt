Test: RandomInt
Test: Math
 PI = 3.141592653589793
sin(0.00) = 0.00
cos(0.00) = 1.00
tan(0.00) = 0.00
sin(0.39) = 0.38
cos(0.39) = 0.92
tan(0.39) = 0.41
sin(0.79) = 0.71
cos(0.79) = 0.71
tan(0.79) = 1.00
sin(1.18) = 0.92
cos(1.18) = 0.38
tan(1.18) = 2.41
sin(1.57) = 1.00
cos(1.57) = 0.00
tan(1.57) = 16331239353195370.00
sin(1.96) = 0.92
cos(1.96) = -0.38
tan(1.96) = -2.41
sin(2.36) = 0.71
cos(2.36) = -0.71
tan(2.36) = -1.00
sin(2.75) = 0.38
cos(2.75) = -0.92
tan(2.75) = -0.41
sin(3.14) = 0.00
cos(3.14) = -1.00
tan(3.14) = -0.00
OK
